auth/auth.go

```diff
@@ -24,37 +26,35 @@ const (
 	ProviderName = "secrets-store-csi-driver-provider-aws"
 )
 
-// Auth is the main entry point to retrieve an AWS session. The caller
+// Auth is the main entry point to retrieve an AWS config. The caller
 // initializes a new Auth object with NewAuth passing the region, namespace, pod name,
 // K8s service account and usePodIdentity flag  (and request context). The caller can then obtain AWS
-// sessions by calling GetAWSSession.
+// config by calling GetAWSConfig.
 type Auth struct {
 	region, nameSpace, svcAcc, podName, preferredAddressType string
 	usePodIdentity                                           bool
 	k8sClient                                                k8sv1.CoreV1Interface
-	stsClient                                                stsiface.STSAPI
-	ctx                                                      context.Context
+	stsClient                                                stscreds.AssumeRoleWithWebIdentityAPIClient
 }
 
-// Factory method to create a new Auth object for an incomming mount request.
+// NewAuth creates an Auth object for an incoming mount request.
 func NewAuth(
-	ctx context.Context,
 	region, nameSpace, svcAcc, podName, preferredAddressType string,
 	usePodIdentity bool,
 	k8sClient k8sv1.CoreV1Interface,
 ) (auth *Auth, e error) {
-	var stsClient stsiface.STSAPI
+	var stsClient *sts.Client
 
 	if !usePodIdentity {
-		// Get an initial session to use for STS calls when using IRSA
-		sess, err := session.NewSession(aws.NewConfig().
-			WithSTSRegionalEndpoint(endpoints.RegionalSTSEndpoint).
```

Comment:
Which STS endpoint are we now using? We want to make sure we are not hitting the global (us-east-1) endpoint by default.